lib/process-manager.ts

```diff
@@ -141,11 +143,28 @@ export class QueryProcessWrapper<T, U> implements ProcessWrapper {
 			const resolve = this.pendingTasks.get(taskId);
 			if (!resolve) throw new Error(`Invalid taskId ${message.slice(0, nlLoc)}`);
 			this.pendingTasks.delete(taskId);
-			resolve(JSON.parse(message.slice(nlLoc + 1)));
+			const resp = this.safeJSON(message.slice(nlLoc + 1));
+			if (resp instanceof Error) {
+				// we'd fail here anyway if it crashed, so at least this way we can log it
+				return false;
```

Comment:
This should probably `resolve` something so we don't have an infinitely waiting Promise somewhere.

Reply:
It might be nice to use [resolve, reject] pairs instead, so we could reject it there.

Reply:
I'll just make it resolve undefined for now, though. Whenever the JSON.parse fails, it's usually undefined anyway.